linera-chain/src/manager/multi.rs

```diff
@@ -81,19 +85,21 @@ impl MultiOwnerManager {
             {
                 ensure!(new_round > *round, ChainError::InsufficientRound(*round));
                 ensure!(
-                    *new_block == *block,
+                    *new_block == *block
+                        || proposal
```

Comment:
It seems to me that when appropriate we should `proposal.validated` in place of `self.locked` entirely here.
Otherwise, where do we check that `proposal.validated` has the same block as `new_block`?

Reply:
Also, I would expect `self.locked` to be updated to the higher round validation certificate (although this may be optional)

Reply:
Good catch!! We need to check they're the same, of course.

Reply:
I'm not sure about updating `locked`: I was trying to keep `check_proposed_block` immutable, so we'd have to update in `create_vote`. It might be clearer if `locked` is _only_ updated in `create_final_vote`.

Reply:
no change here?